### PR TITLE
Bug 1914941: VM Template rootdisk is not deleted after fetching default disk bus

### PR DESCRIPTION
**Pull request by glekner**

### CODE REVIEW

File: frontend/packages/kubevirt-plugin/src/components/create-vm-wizard/redux/initial-state/storage-tab-initial-state.ts [hunk 1]
```diff
@@ -240,14 +237,12 @@ export const getNewProvisionSourceStorage = (state: any, id: string): VMWizardSt
...
   const iCommonTemplates = iGetLoadedCommonData(state, id, VMWizardProps.commonTemplates);
   const iTemplate = iCommonTemplates && iGetRelevantTemplate(iCommonTemplates, relevantOptions);
-  // eslint-disable-next-line no-template-curly-in-string
-  const tmpDiskBus = DiskBus.fromString(iGetCommonTemplateDiskBus(iTemplate, '${NAME}'));
 
   const initialData = getInitialData(state, id);
   const { source } = initialData;
   const storagesUpdate = getStorages(state, id);
   const rootStorage = storagesUpdate.find((s) => s.disk.bootOrder === 1) || storagesUpdate[0];
-  const diskBus = tmpDiskBus || new DiskWrapper(rootStorage?.disk).getDiskBus();
```
Comment (suomiy) [inline frontend/packages/kubevirt-plugin/src/components/create-vm-wizard/redux/initial-state/storage-tab-initial-state.ts:250]: what is wrong with this part?

**Review by suomiy — CHANGES_REQUESTED** [review]

**Comment by suomiy** [comment]
/hold

@metalice can you please wait with the lgtms a bit in new PRs?

File: frontend/packages/kubevirt-plugin/src/hooks/use-update-data-volume.ts [hunk 2]
```diff
@@ -34,10 +38,26 @@ export const useUpdateStorages = (reduxID) => {
...
     updateStorage(rootDisk);
   }
+
+  if (templateDisk) {
+    const templateBus = templateDisk?.disk?.disk?.bus;
+    if (templateBus) {
+      rootDisk.disk.disk.bus = templateBus;
+      updateStorage(rootDisk);
+    }
+    removeStorage(templateDisk);
```
[the remark this answers is not in the document]
Comment (suomiy) [reply frontend/packages/kubevirt-plugin/src/hooks/use-update-data-volume.ts:61]: Also, as I noted [here](https://github.com/openshift/console/pull/7749) we probably shouldn't have the decision logic scattered everywhere but just in updaters. We can open a discussion regarding this topic if you want (cc @metalice )

+1 for moving this from storage-tab - which doesn't have to be even rendered.

**Comment by glekner** [comment]
@suomiy I think I fixed it

File: frontend/packages/kubevirt-plugin/src/components/create-vm-wizard/redux/initial-state/storage-tab-initial-state.ts (hunk 1 above)
Comment (glekner) [reply frontend/packages/kubevirt-plugin/src/components/create-vm-wizard/redux/initial-state/storage-tab-initial-state.ts]: reverted

**Comment by suomiy** [comment]
+1 simpler and cleaner
/lgtm

File: frontend/packages/kubevirt-plugin/src/components/create-vm-wizard/redux/state-update/prefill-vm-template-state-update.ts [hunk 3]
```diff
@@ -209,7 +210,7 @@ export const prefillVmTemplateUpdater = ({ id, dispatch, getState }: UpdateOptio
     const standaloneDataVolumeLookup = createBasicLookup<V1alpha1DataVolume>(dataVolumes, getName);
 
     const vmDisks = commonTemplateName
-      ? getDisks(vm).filter((d) => d.name !== 'rootdisk')
+      ? getDisks(vm).filter((d) => d.name !== VM_TEMPLATE_NAME_PARAMETER)
```
Comment (glekner) [inline frontend/packages/kubevirt-plugin/src/components/create-vm-wizard/redux/state-update/prefill-vm-template-state-update.ts:213]: @yaacov @metalice please note this is the correct fix, we should stick to updaters.

**Pull request merged**